__tests__/app.test.js

```diff
@@ -77,6 +77,93 @@ describe("/api/articles", () => {
+                .expect(200)
+                .then(({ body: { articles } }) => {
+                    expect(articles).toBeSortedBy("title", {
+                        descending: true
+                    });
+
+                    return request(app)
+                        .get("/api/articles?sort_by=created_at")
+                        .expect(200);
+                })
+                .then(({ body: { articles } }) => {
+                    expect(articles).toBeSortedBy("created_at", {
+                        descending: true
+                    });
+                    return request(app)
+                        .get("/api/articles?sort_by=votes")
+                        .expect(200);
+                })
+                .then(({ body: { articles } }) => {
+                    expect(articles).toBeSortedBy("votes", {
+                        descending: true
+                    });
+                });
+        });
+        test("GET - 200: valid order query sets sorting order (defaults to descending)", () => {
+            return request(app)
+                .get("/api/articles?order=asc")
+                .expect(200)
+                .then(({ body: { articles } }) => {
+                    expect(articles).toBeSortedBy("created_at", {
+                        descending: false
+                    });
+                });
+        });
+        test("GET - 200: valid combination of all three queries works as desired, and uppercase is ignored", () => {
+            return request(app)
+                .get("/api/articles?sort_by=AUTHOR&order=ASC&topic=MITCH")
+                .expect(200)
+                .then(({ body: { articles } }) => {
+                    expect(articles.length).toBe(11);
+                    expect(articles).toBeSortedBy("author", {
+                        descending: false
+                    });
+                });
+        });
+        test("GET - 400: invalid order query returns error", () => {
+            return request(app)
+                .get("/api/articles?order=gobbledigook")
+                .expect(400)
+                .then(({ body: { msg } }) => {
+                    expect(msg).toBe("invalid querystring");
+                });
+        });
+        test("GET - 400: invalid sort_by query returns error", () => {
+            return request(app)
+                .get("/api/articles?sort_by=gobbledigook")
+                .expect(400)
+                .then(({ body: { msg } }) => {
+                    expect(msg).toBe("invalid querystring");
+                });
+        });
+        test.skip("GET - 404: invalid topic query returns error", () => {
+            return request(app)
+                .get(
+                    "/api/articles?topic=Beethoven'sSeventhSymphonyIsTheBestOne"
+                )
+                .expect(404)
+                .then(({ body: { msg } }) => {
+                    expect(msg).toBe("topic not found");
+                });
+        });
+    });
```

Comment:
So we do want to check for querying a topic which has no responses (it's not an invalid topic, there just might not be any articles yet!)

This would be a 200 returning an empty array